tests/baselines/reference/mappedTypeWithAsClauseAndLateBoundProperty2.js

```diff
@@ -4,3 +4,152 @@ export const thing = (null as any as { [K in keyof number[] as Exclude<K, "lengt
+    };
+    some: (predicate: (value: number, index: number, array: number[]) => unknown, thisArg?: any) => boolean;
+    forEach: (callbackfn: (value: number, index: number, array: number[]) => void, thisArg?: any) => void;
+    map: <U>(callbackfn: (value: number, index: number, array: number[]) => U, thisArg?: any) => U[];
+    filter: {
+        <S_1 extends number>(predicate: (value: number, index: number, array: number[]) => value is S_1, thisArg?: any): S_1[];
+        (predicate: (value: number, index: number, array: number[]) => unknown, thisArg?: any): number[];
+    };
+    reduce: {
+        (callbackfn: (previousValue: number, currentValue: number, currentIndex: number, array: number[]) => number): number;
+        (callbackfn: (previousValue: number, currentValue: number, currentIndex: number, array: number[]) => number, initialValue: number): number;
+        <U_1>(callbackfn: (previousValue: U_1, currentValue: number, currentIndex: number, array: number[]) => U_1, initialValue: U_1): U_1;
+    };
+    reduceRight: {
+        (callbackfn: (previousValue: number, currentValue: number, currentIndex: number, array: number[]) => number): number;
+        (callbackfn: (previousValue: number, currentValue: number, currentIndex: number, array: number[]) => number, initialValue: number): number;
+        <U_2>(callbackfn: (previousValue: U_2, currentValue: number, currentIndex: number, array: number[]) => U_2, initialValue: U_2): U_2;
+    };
+    find: {
+        <S_2 extends number>(predicate: (value: number, index: number, obj: number[]) => value is S_2, thisArg?: any): S_2;
+        (predicate: (value: number, index: number, obj: number[]) => unknown, thisArg?: any): number;
+    };
+    findIndex: (predicate: (value: number, index: number, obj: number[]) => unknown, thisArg?: any) => number;
+    fill: (value: number, start?: number, end?: number) => number[];
+    copyWithin: (target: number, start: number, end?: number) => number[];
+    entries: () => IterableIterator<[number, number]>;
+    keys: () => IterableIterator<number>;
+    values: () => IterableIterator<number>;
+    includes: (searchElement: number, fromIndex?: number) => boolean;
+    flatMap: <U_3, This = undefined>(callback: (this: This, value: number, index: number, array: number[]) => U_3 | readonly U_3[], thisArg?: This) => U_3[];
+    flat: <A, D extends number = 1>(this: A, depth?: D) => FlatArray<A, D>[];
+    [Symbol.iterator]: () => IterableIterator<number>;
+    [Symbol.unscopables]: () => {
+        copyWithin: boolean;
+        entries: boolean;
+        fill: boolean;
+        find: boolean;
+        findIndex: boolean;
+        keys: boolean;
+        values: boolean;
+    };
+};
+
+
+//// [DtsFileErrors]
+
+
+tests/cases/compiler/mappedTypeWithAsClauseAndLateBoundProperty2.d.ts(24,118): error TS2526: A 'this' type is available only in a non-static member of a class or interface.

```

Comment:
This is a little bit worrying though. It doesn't exactly look like a new problem though. It's just accidental that by making this thing serializable I now emit this error. I think that this should be handled separately. WDYT?

Reply:
Definitely it can be handled separately; I can't tell if it's a bug or not that `this` is emitted as a return type of an object literal, but it's worth investigating.